composer/profiler/torch_profiler.py

```diff
@@ -202,28 +252,116 @@ def handler_fn(prof: torch.profiler.profiler.profile):
             assert state.profiler is not None
 
             timestamp = state.timestamp
-
-            trace_file_name = os.path.join(
-                folder_name,
-                format_name_with_dist_and_time(self.filename, run_name=state.run_name, timestamp=timestamp),
-            )
-            trace_file_dirname = os.path.dirname(trace_file_name)
-            if trace_file_dirname:
-                os.makedirs(trace_file_dirname, exist_ok=True)
-            prof.export_chrome_trace(trace_file_name)
-            state.profiler.record_chrome_json_trace_file(trace_file_name)
-            if self.remote_file_name is not None:
-                trace_remote_file_name = format_name_with_dist_and_time(self.remote_file_name,
-                                                                        run_name=state.run_name,
-                                                                        timestamp=timestamp)
-                trace_remote_file_name = trace_remote_file_name.lstrip('/')
-                logger.upload_file(remote_file_name=trace_remote_file_name,
-                                   file_path=trace_file_name,
-                                   overwrite=self.overwrite)
+            if not (self.filename is None and self.memory_filename is not None):
```

Comment:
```suggestion
            if self.filename is not None:
```

We should output traces if self.filename is not None (independent of self.memory_filename)